src/main/java/com/amplitude/Event.java

```diff
@@ -240,11 +240,18 @@ public JSONObject toJsonObject() {
       event.put(
           "user_properties",
           (userProperties == null) ? new JSONObject() : truncate(userProperties));
-      event.put("price", price);
-      event.put("quantity", quantity);
-      event.put("revenue", revenue);
-      event.put("productId", productId);
-      event.put("revenueType", revenueType);
+
+      boolean shouldLogRevenueProps = (revenue != null || (price != null && quantity > 0));
```

Comment:
Could we add a simple test case for this?

Reply:
Sure, will add the test.